tests/test_visitors/test_ast/test_classes/test_methods/test_useless_overwriting_method.py

```diff
@@ -0,0 +1,337 @@
+    visitor.run()
+
+    assert_errors(visitor, [])
+
+
+@pytest.mark.parametrize('code', [
+    _regular_method_detailed,
+])
+@pytest.mark.parametrize('statements', valid_statements)
+@pytest.mark.parametrize('method_args', valid_method_args)
+@pytest.mark.parametrize('super_args', invalid_super_args)
+def test_useful_due_to_invalid_super_args(
+    assert_errors,
+    parse_ast_tree,
+    mode,
+    code,
+    statements,
+    super_args,
+    method_args,
+    default_options,
+):
+    """Testing situations with useful overwriting due to invalid super args."""
+    formatted_code = mode(code.format(
+        decorator='',
+        args_definition=method_args.definition,
+        statements=statements,
+        super_args=super_args,
+        method_name='function',
+        args_invocation=method_args.invocation,
+    ))
+    tree = parse_ast_tree(formatted_code)
+
+    visitor = WrongMethodVisitor(default_options, tree=tree)
+    visitor.run()
+
+    assert_errors(visitor, [])
+
+
+@pytest.mark.parametrize('code', [
+    _regular_method_detailed,
+])
+@pytest.mark.parametrize('statements', valid_statements)
+@pytest.mark.parametrize('method_args', valid_method_args)
+@pytest.mark.parametrize('super_args', valid_super_args)
+def test_useful_due_to_invalid_method(
+    assert_errors,
+    parse_ast_tree,
+    mode,
+    code,
+    statements,
+    super_args,
+    method_args,
+    default_options,
+):
+    """Testing situations with useful overwriting due to invalid method."""
+    formatted_code = mode(code.format(
+        decorator='',
+        args_definition=method_args.definition,
+        statements=statements,
+        super_args=super_args,
+        method_name='invalid_function',
+        args_invocation=method_args.invocation,
+    ))
+    tree = parse_ast_tree(formatted_code)
+
+    visitor = WrongMethodVisitor(default_options, tree=tree)
+    visitor.run()
+
+    assert_errors(visitor, [])
+
+
+@pytest.mark.parametrize('code', [
+    _regular_method_detailed,
+])
+@pytest.mark.parametrize('statements', valid_statements)
+@pytest.mark.parametrize('method_args', invalid_method_args)
+@pytest.mark.parametrize('super_args', valid_super_args)
+def test_useful_due_to_invalid_method_args(
+    assert_errors,
+    parse_ast_tree,
+    mode,
+    code,
+    statements,
+    super_args,
+    method_args,
+    default_options,
+):
+    """Testing situations with useful overwriting due to invalid method args."""
+    formatted_code = mode(code.format(
+        decorator='',
+        args_definition=method_args.definition,
+        statements=statements,
+        super_args=super_args,
+        method_name='function',
+        args_invocation=method_args.invocation,
+    ))
+    tree = parse_ast_tree(formatted_code)
+
+    visitor = WrongMethodVisitor(default_options, tree=tree)
+    visitor.run()
+
+    assert_errors(visitor, [])
+
+
+@pytest.mark.parametrize('code', [
+    _regular_method_short,
```

Comment:
Can you please explain why do we need this `_regular_method_short`?

Reply:
do you mean why is it a variable and not just inline?

Reply:
I mean: how `_regular_method_short` is different from `_regular_method_detailed`?

Reply:
in `_regular_method_detailed` `self` is not parameterized but I can actually change that and drop `_regular_method_short` if you want

Reply:
in `_regular_method_detailed` `self` is not parameterized but I can actually change that and `drop _regular_method_short if you want`

Reply:
I have tried to use `_regular_method_detailed` where `_regular_method_short` was used but it's more complicated cause `detailed` has `return super(...)` whereas for general case I want to check other statements.

So I would stick with current approach if you don't mind.

Reply:
👍